frame/lending/src/lib.rs

```diff
@@ -627,12 +631,14 @@ pub mod pallet {
 		#[transactional]
 		pub fn create_market(
 			origin: OriginFor<T>,
-			input: Validated<CreateInputOf<T>, (MarketModelValid, CurrencyPairIsNotSame)>,
+			input: Validated<
+				CreateInputOf<T>,
+				(MarketModelValid, CurrencyPairIsNotSame, AssetIsSupportedByOracle<T::Oracle>),
+			>,
 			keep_alive: bool,
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
-			let input = input.value();
-			let pair = input.currency_pair;
+			let pair = input.clone().value().currency_pair;
```

Comment:
not sure, it was working without clone. not big deal for create, but still can avoid clone?